cpp/ql/test/experimental/query-tests/Security/CWE/CWE-758/semmle/tests/test.cpp

```diff
@@ -0,0 +1,218 @@
+#define NULL (0)
+typedef unsigned long size_t;
+typedef struct {} FILE;
+FILE *ef;
+void *malloc(size_t size);
+size_t strlen(const char *str);
+void free(void *ptr);
+unsigned long copy_from_user (void * to, void * from, unsigned long n);
+int fread(char *buf, int size, int count, FILE *fp);
+int getSize3();
+int getLen3(char *s);
+
+int getSize2()
+{
+  int m;
+  m = getSize3();
+  if (m<0)  return 0;
+  return m;
+}
+
+int getSize1()
+{
+  int m;
+  m = getSize3();
+  if (m<0) m=0;
+  return m;
+}
+int getLen4() {
+  char c[1];
+  fread(c, 1, 1, ef);
```

Comment:
`ef` is uninitialized at this point. Maybe you should take `ef` as a parameter instead?

Reply:
and fixed it, through a function call.